apps/api/v1/pages/api/bookings/_get.ts

```diff
@@ -130,25 +137,30 @@ type GetAdminArgsType = {
   requestedUserIds: number[];
   userId: number;
 };
+
 /**
  * Constructs the WHERE clause for Prisma booking findMany operation.
  *
  * @param userId - The ID of the user making the request. This is used to filter bookings where the user is either the host or an attendee.
  * @param attendeeEmails - An array of emails provided in the request for filtering bookings by attendee emails, used in case of Admin calls.
  * @param userIds - An array of user IDs to be included in the filter. Defaults to an empty array, and an array of user IDs in case of Admin call containing it.
  * @param userEmails - An array of user emails to be included in the filter if it is an Admin call and contains userId in query parameter. Defaults to an empty array.
+ * @param upcomingOnly - A boolean flag to filter only upcoming bookings.
  *
  * @returns An object that represents the WHERE clause for the findMany/findUnique operation.
  */
 function buildWhereClause(
   userId: number | null,
   attendeeEmails: string[],
   userIds: number[] = [],
-  userEmails: string[] = []
+  userEmails: string[] = [],
+  upcomingOnly: boolean
```

Comment:
Like I described in other comment - we can pass 1 prop `status: Status` and then  if status === "upcoming" we add the where clause.